src/analysis/asymptotics/superpolynomial_decay.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import analysis.asymptotics.asymptotics
+import analysis.asymptotics.specific_asymptotics
+import data.polynomial.eval
+
+/-!
+# Super-Polynomial Function Decay
+
+This file defines a predicate `asymptotics.superpolynomial_decay f` for a function satisfying
+  one of following equivalent definitions (The definition is in terms of the first condition):
+
+* `f` is `O(x ^ c)` for all (or equivalently sufficiently small) integers `c`
+* `f` is `O(p(x)⁻¹)` for all (or equivalently sufficiently large) polynomials `p`
+* `p(x) * f` is bounded for all polynomials `p`
+* `p(x) * f` tends to `𝓝 0` for all polynomials `p`
```

Comment:
You don't actually prove this equivalence, do you? You should explain this properly / note this as a future project, etc. Or possibly do it!

I will admit I still wish the definition was written as "`p(x) * f` is bounded for all polynomials `p`", because we can state this without needing to say anything about asymptotics, or even fields.

Reply:
I ended up just proving the equivalence between all these conditions in terms of `n ^ c`, leaving the polynomial versions of the lemmas unproven for now.

I also started refactoring to try and use the equivalent definition from `superpolynomial_decay_iff_tendsto_zero`, but there's still a lot left to work through so it may be better to leave for a future PR to make that change